Panic using TcpOutput with tls=true

```
panic: runtime error: invalid memory address or nil pointer dereference
    panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x618ce5]

goroutine 9 [running]:
runtime.panic(0x8357e0, 0xfdb088)
    /usr/local/go/src/pkg/runtime/panic.c:266 +0xb6
crypto/tls.(*Conn).Close(0x0, 0x0, 0x0)
    /usr/local/go/src/pkg/crypto/tls/conn.go:924 +0x115
github.com/mozilla-services/heka/plugins/tcp.func·004()
    /home/ubuntu/heka/build/heka/src/github.com/mozilla-services/heka/plugins/tcp/tcp_output.go:269 +0xa7
runtime.panic(0x8357e0, 0xfdb088)
    /usr/local/go/src/pkg/runtime/panic.c:248 +0x106
crypto/tls.(*Conn).Close(0x0, 0x0, 0x0)
    /usr/local/go/src/pkg/crypto/tls/conn.go:924 +0x115
github.com/mozilla-services/heka/plugins/tcp.(*TcpOutput).StreamOutput(0xc21005b2c0, 0xc21005b580, 0xc210037d20, 0xc210037d80)
    /home/ubuntu/heka/build/heka/src/github.com/mozilla-services/heka/plugins/tcp/tcp_output.go:331 +0x56a
created by github.com/mozilla-services/heka/plugins/tcp.(*TcpOutput).Run
    /home/ubuntu/heka/build/heka/src/github.com/mozilla-services/heka/plugins/tcp/tcp_output.go:369 +0x1b7
```

Full output here: http://pastebin.mozilla.org/4352145

Relevant config snippets:
### Server

```
[ExternalTcpInput]
type = "TcpInput"
address = "x.x.x.x:5565"
parser_type = "message.proto"
decoder = "ProtobufDecoder"
use_tls = true

    [ExternalTcpInput.tls]
    cert_file = "/usr/share/heka/tls/cert.crt"
    key_file = "/usr/share/heka/tls/cert.key"
    client_auth = "RequireAndVerifyClientCert"
    prefer_server_ciphers = true
```
### Client

```
[TcpOutput]
address = "myserver:5565"
message_matcher = "TRUE"
use_tls = true

    [TcpOutput.tls]
    server_name = "myhost"
    cert_file = "/usr/share/heka/tls/cert.crt"
    key_file = "/usr/share/heka/tls/cert.key"
```

Removing `use_tls` from the client fixes this issue (although the client can no longer connect).

--- a/plugins/tcp/tcp_output.go
+++ b/plugins/tcp/tcp_output.go
@@ -237,6 +237,9 @@ func (t *TcpOutput) sendRecord(record []byte) (err error) {
 	var n int
 	if t.connection == nil {
 		if err = t.connect(); err != nil {
+			// Explicitly set t.connection to nil b/c Go, see
+			// http://golang.org/doc/faq#nil_error.
+			t.connection = nil
 			return
 		}
 	}

